store/store.go

```diff
@@ -1453,7 +1471,11 @@ func (s *Store) Download(ctx context.Context, name string, targetPath string, do
 		return err
 	}
 
-	return w.Sync()
+	if err := w.Sync(); err != nil {
+		return err
+	}
+
+	return s.cacher.Put(targetPath)
```

Comment:
I think we are not really testing integration of the cache within store code, i.e. not testing that Lookup, Get and Put are actually used in ``store.Download(..)`` when they should, are we? Would it be possible to have a test for that?

Reply:
Thanks, very good point. I added tests for this now (one for cache hit, one for cache miss).